types/tx/types.go

```diff
@@ -100,8 +101,15 @@ func (t *Tx) GetSigners() []sdk.AccAddress {
 // UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
 func (t *Tx) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
 	if t.Body != nil {
-		return t.Body.UnpackInterfaces(unpacker)
+		if err := t.Body.UnpackInterfaces(unpacker); err != nil {
+			return err
+		}
 	}
+
+	if t.AuthInfo != nil {
+		return t.AuthInfo.UnpackInterfaces(unpacker)
+	}
```

Comment:
Shouldn't we return error if both `Body == nil` or `AuthInfo==nil`?

Reply:
UnpackInterfaces just unpacks all `Any`s in the struct's nested fields. If there are no such fields, then it should do nothing, I believe it's better not to error here.

Reply:
When a `Body` can be empty? Is this expected?

Reply:
Yeah I see what you mean. The `Body` of a Tx will never be empty indeed.

My idea is, `UnpackInterfaces` should not care about the logic of the struct, it should not "know" that `Body` cannot be nil, it should just unpack recursively, when possible.